test/integration/extensions/webhook/cloudprovider/cloudprovider_suite_test.go

```diff
@@ -0,0 +1,160 @@
+
+var (
+	ctx = context.TODO()
+	log logr.Logger
+
+	restConfig *rest.Config
+	testEnv    *envtest.Environment
+	testClient client.Client
+
+	testNamespace *corev1.Namespace
+	providerName  = "test-provider"
+)
+
+var _ = BeforeSuite(func() {
+	logf.SetLogger(logger.MustNewZapLogger(logger.DebugLevel, logger.FormatJSON, zap.WriteTo(GinkgoWriter)))
+	log = logf.Log.WithName(testID)
+
+	By("starting test environment")
+	testEnv = &envtest.Environment{
+		CRDInstallOptions: envtest.CRDInstallOptions{
+			Paths: []string{
+				filepath.Join("..", "..", "..", "..", "..", "example", "seed-crds", "10-crd-extensions.gardener.cloud_clusters.yaml"),
+			},
+		},
+		ErrorIfCRDPathMissing: true,
+	}
+
+	var err error
+	restConfig, err = testEnv.Start()
+	Expect(err).NotTo(HaveOccurred())
+	Expect(restConfig).NotTo(BeNil())
+
+	DeferCleanup(func() {
+		By("stopping test environment")
+		Expect(testEnv.Stop()).To(Succeed())
+	})
+
+	By("creating test client")
+	testClient, err = client.New(restConfig, client.Options{Scheme: kubernetes.SeedScheme})
+	Expect(err).NotTo(HaveOccurred())
+
+	By("creating test namespace")
+	testNamespace = &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			// create dedicated namespace for each test run, so that we can run multiple tests concurrently for stress tests
+			GenerateName: testID + "-",
+			Labels: map[string]string{
+				v1beta1constants.LabelShootProvider: providerName,
+			},
+		},
+	}
+	Expect(testClient.Create(ctx, testNamespace)).To(Succeed())
+	log.Info("Created Namespace for test", "namespaceName", testNamespace.Name)
+
+	DeferCleanup(func() {
+		By("deleting test namespace")
+		Expect(testClient.Delete(ctx, testNamespace)).To(Or(Succeed(), BeNotFoundError()))
+	})
+
+	By("setup manager")
+	mgr, err := manager.New(restConfig, manager.Options{
+		Port:               testEnv.WebhookInstallOptions.LocalServingPort,
+		Host:               testEnv.WebhookInstallOptions.LocalServingHost,
+		CertDir:            testEnv.WebhookInstallOptions.LocalServingCertDir,
+		Scheme:             kubernetes.SeedScheme,
+		MetricsBindAddress: "0",
+		Namespace:          testNamespace.Name,
+	})
+	Expect(err).NotTo(HaveOccurred())
+
+	By("registering webhook")
+	Expect(addTestWebhookToManager(mgr, false)).To(Succeed())
+
+	By("starting manager")
+	mgrContext, mgrCancel := context.WithCancel(ctx)
+
+	go func() {
+		defer GinkgoRecover()
+		Expect(mgr.Start(mgrContext)).To(Succeed())
+	}()
+
+	DeferCleanup(func() {
+		By("stopping manager")
+		mgrCancel()
+	})
+})
+
+func addTestWebhookToManager(mgr manager.Manager, enableObjectSelector bool) error {
+	switchOptions := webhookcmd.NewSwitchOptions(
+		webhookcmd.Switch("cloudprovider", func(mgr manager.Manager) (*extensionswebhook.Webhook, error) {
+			return cloudprovider.New(mgr, cloudprovider.Args{
+				Provider:             providerName,
+				Mutator:              cloudprovider.NewMutator(log, testcloudprovider.NewEnsurer(log)),
+				EnableObjectSelector: enableObjectSelector,
```

Comment:
Can we use `EnableObjectSelector=true` and label the secret accordingly?
This is closer to the production code.